docs/arch/adr-59.md

```diff
@@ -0,0 +1,21 @@
+# ADR 58: Incremental Static Regeneration for project pages
+
+## Status
+Accepted
+
+## Context
+When the first Engaging Crowds projects launched, project pages were built dynamically with `getServerSideProps`. Page builds were slow, with pages sometimes taking several seconds to load in the browser ([#2741](https://github.com/zooniverse/front-end-monorepo/issues/2741).)
+
+## Decision
+Use [Incremental Static Regeneration](https://nextjs.org/docs/pages/building-your-application/data-fetching/incremental-static-regeneration) (ISR) to build project pages. ISR is a feature of Next.js, which allows page HTML and JSON to be built and cached on demand, then served with `stale-while-revalidate` cache control headers. While a page is building, visitors will get stale, cached content, resulting in a faster experience.
+
+## Consequences
+- Page builds were switched from `getServerSideProps` to `getStaticProps`, with a revalidation time of 60s. Revalidation can be set individually for each page route.
+- `getStaticProps` can still be slow, when it runs, so data-fetching from the Panoptes API may need further optimisation ([#3341](https://github.com/zooniverse/front-end-monorepo/issues/3341).)
```

Comment:
Could this also be a cold-start issue when the function first runs? Similar to https://github.com/vercel/next.js/discussions/12447.

Reply:
I'm not sure about that, as the slowness persists after the first call. Here's a screenshot of the network tab for Deciphering Secrets, showing slow requests for JSON page props (marked with a turtle icon in Firefox dev tools.)

https://www.zooniverse.org/projects/profdrrogerlouismartinez-davila/deciphering-secrets-unlocking-the-manuscripts-of-medieval-spain/classify/workflow/19276?demo=true&env=production

<img width="600" alt="Network requests for workflow and subject set page props on Deciphering Secrets. Several prefetch requests for JSON props are slow, taking ~0.5s to download." src="https://github.com/zooniverse/front-end-monorepo/assets/59547/df4c34fd-5a69-4fb2-916b-23ec758b85f7">

Next.js prefetching probably doesn't help here, in that page props are fetched whenever you hover over a link. `getStaticProps` will run for those links if the cache is stale.